setup.py

```diff
@@ -52,19 +67,45 @@ def main():
     class SetupStepFailedError(ValueError):
         pass
 
-    def getInputPath(prompt):
-        import shlex
-
+    def getOutputPath(prompt, default):
+        prompt_cli = prompt + f' ({default=}): '
         while True:
-            path = ''
-            while len(path) == 0:
-                path = input(prompt + ' ')
+            if gui_mode:
+                filename = filedialog.asksaveasfilename(confirmoverwrite=False, initialfile=default, title=prompt, defaultextension=".bin", filetypes=(("BIN file", ""), ))
+            else:
+                filename = getResourcePath(input(prompt_cli) or default)
+            if not filename:
+                sys.exit('Setup aborted by user.')
+            elif os.path.isfile(filename):
+                print(f'{filename!r} already exists!')
+            else:
+                try:
+                    open(filename, 'w+').close()
+                    os.unlink(filename)
+                    return filename
+                except OSError:
+                    print(f'{filename!r} is not writable!')
 
-            display_path = shlex.split(path)[0]
-            result = getResourcePath(display_path)
-            if os.path.isfile(result):
-                return result
-            print(f'{display_path!r} is not a valid file!')
+    def getInputPath(prompt):
+        if gui_mode:
+            while True:
+                filename = filedialog.askopenfilename(title=prompt)
```

Comment:
Can this throw?

Reply:
Also, can this be filtered for `.bin` and `.iso` files so the user cannot pick e.g. the cuesheet?

Technically `.bin` is the only valid one but in practice many/most PS1 `.iso` files are just renamed binfiles.

Reply:
> Also, can this be filtered for `.bin` and `.iso` files so the user cannot pick e.g. the cuesheet?
> 
> Technically `.bin` is the only valid one but in practice many/most PS1 `.iso` files are just renamed binfiles.

Done! I kinda just forgot after putting the equivalent in the save dialog.